### ISSUE TITLE
Cauchy type mismatch

### ISSUE THREAD

**Issue by JeffFessler** [JuliaLinearAlgebra/SpecialMatrices.jl#46]
The Cauchy type claims it is a subtype of `AbstractArray{T}` but the `T` is incorrect in general:

```julia
julia> C = Cauchy([1,2,3])
3×3 Cauchy{Int64}:
 0.5       0.333333  0.25
 0.333333  0.25      0.2
 0.25      0.2       0.166667
```

Here the Cauchy "type" is `Int64` but the elements of the matrix are `Float64`.
I don't have time to fix this right now but I am logging the issue for future work.
A related PR #17 does not appear to solve this issue BTW.

**Comment by JeffFessler** [comment]
Will be fixed by the updated #17.

**Comment by JeffFessler** [comment]
Fixed by #17
